src/bun.js/webcore.zig

```diff
@@ -13,6 +14,279 @@ pub const Lifetime = enum {
+
+            if (byte == delimiter) {
+                return;
+            }
+
+            try array_list.append(byte);
+        }
+    }
+
+    /// https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#dom-prompt
+    pub fn call(
+        // this
+        _: void,
+        ctx: JSC.C.JSContextRef,
+        // function
+        _: JSC.C.JSObjectRef,
+        // thisObject
+        _: JSC.C.JSObjectRef,
+        arguments: []const JSC.C.JSValueRef,
+        _: JSC.C.ExceptionRef,
+    ) JSC.C.JSValueRef {
+        var stdout = std.io.getStdOut();
+        const has_message = arguments.len != 0 and arguments[0] != null;
+        // 4. Set default to the result of optionally truncating default.
+        // *  We don't really need to do this.
+        const default = if (arguments.len >= 2 and arguments[1] != null) arguments[1] else JSC.JSValue.jsNull().asObjectRef();
+
+        if (has_message) {
+            // 2. Set message to the result of normalizing newlines given message.
+            // *  Not pertinent to a server runtime so we will just let the terminal handle this.
+
+            // 3. Set message to the result of optionally truncating message.
+            // *  Not necessary so we won't do it.
+            const message = arguments[0].?.value().toSlice(ctx.ptr(), bun.default_allocator);
+
+            stdout.writeAll(message.slice()) catch {
+                // 1. If we cannot show simple dialogs for this, then return null.
+                return JSC.JSValue.jsNull().asObjectRef();
+            };
+        }
+
+        // 4. Set default to the result of optionally truncating default.
+
+        // 5. Show message to the user, treating U+000A LF as a line break,
+        //    and ask the user to either respond with a string value or
+        //    abort. The response must be defaulted to the value given by
+        //    default.
+        stdout.writeAll(if (has_message) " " else "Prompt ") catch {
+            // 1. If we cannot show simple dialogs for this, then return false.
+            return JSC.JSValue.jsBoolean(false).asObjectRef();
+        };
+
+        // 6. Invoke WebDriver BiDi user prompt opened with this, "prompt" and message.
+        // *  Not relevant in a server context.
+
+        // 7. Pause while waiting for the user's response.
+        var stdin = std.io.getStdIn();
+        var reader = stdin.reader();
+
+        const first_byte = reader.readByte() catch {
+            // 8. Let result be null if the user aborts, or otherwise the string
+            //    that the user responded with.
+            return JSC.JSValue.jsNull().asObjectRef();
+        };
+
+        if (first_byte == '\n') {
+            // 8. Let result be null if the user aborts, or otherwise the string
+            //    that the user responded with.
+            return default;
+        }
+
+        var message = std.ArrayList(u8).initCapacity(bun.default_allocator, 1) catch {
```

Comment:
Instead of a dynamic allocation here, let's do a `StackFallbackAllocator` with 2048 bytes